team-01/src/main/java/ubc/cosc322/MonteCarloPlayer.java

```diff
@@ -130,5 +128,24 @@ public int getVisits() {
     public AmazonsAction getAction() {
       return action;
     }
+
+    // TODO
+    private double getUCB() {
+      // EXPLORATION_FACTOR = constant defined at the top of the class.
+      // If we hit 0, then the unvisited node should return infinity.
+      if (this.getVisits() == 0){
+        return Double.MAX_VALUE;
+      }
+
+      // uct = v = total score / number of visits == avg value of the state.
+      float uct =  this.wins / this.visits;
+
+      // apply the UCB1 function for that state
+      if (this.parent != null) {
+        uct += EXPLORATION_FACTOR * Math.sqrt(Math.log(this.parent.getVisits()) / this.getVisits());
+      }
```

Comment:
Similar note again: you use the getter in two places in this function, and a direct reference in two other places. This should be consistent.

Also, I forgot to mention it in my last review, but using `this` is unnecessary here, and adds confusion.